library/std/src/sys/net/connection/socket/unix.rs

```diff
@@ -430,6 +434,16 @@ impl Socket {
         setsockopt(self, libc::SOL_SOCKET, SO_LINGER, linger)
     }
 
+    #[cfg(target_os = "cygwin")]
+    pub fn set_linger(&self, linger: Option<Duration>) -> io::Result<()> {
+        let linger = libc::linger {
+            l_onoff: linger.is_some() as libc::c_ushort,
+            l_linger: linger.unwrap_or_default().as_secs() as libc::c_ushort,
```

Comment:
should this be a `try_into().map_error(|_| libc::ERANGE)`?

Reply:
It's just an imitation of the code above, for other platforms. Other platforms also use `as`.